test/BUILD

```diff
@@ -1,49 +1,69 @@
-[sh_test(
-    name = test,
-    # top showed me 50-400mByte of usage, so "enormous" for
-    # long running tests, but the OpenROAD tests are generally
-    # "small" by design.
-    #
-    # https://bazel.build/reference/be/common-definitions#test.size
-    size = "small",
-    timeout = "moderate",
-    srcs = ["bazel_test.sh"],
-    args = [],
-    data = [
-        "//:openroad",
-    ] +
+load("//test:regression.bzl", "regression_test")
+
+exports_files(
+    [
+        "bazel_test.sh",
+        "regression_test.sh",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "regression_resources",
     # TODO Refine glob later.
     #
     # This is a very broad glob, but most of the data in size
     # comes from .lib files and we can't know which pdk files
     # to include for a test given the name only.
-    glob([
-        "**/*",
-    ]),
-    env = {
-        "TEST_NAME_BAZEL": test,
-        "OPENROAD_EXE": "$(location //:openroad)",
-    },
-) for test in [
-    # From CMakeLists.txt or_integration_tests(TESTS
+    srcs = glob(["**/*"]),
+    visibility = ["//visibility:public"],
+)
+
+# From CMakeLists.txt or_integration_tests(TESTS
+COMPULSORY_TESTS = [
     "error1",
     "get_core_die_areas",
     "place_inst",
     "upf_test",
     "upf_aes",
-    # Bazel OpenROAD is not compiled with Python support yet
-    #
-    # "timing_api",
-    # "timing_api_2",
-    # "timing_api_3",
-    # "timing_api_4",
-    # "two_designs",
-    #
-    # TODO There are tests in this folder that were not enabled in
-    # CMakeLists.txt, presumably because they are long running.
-    # These should be added with tags = ["manual"], i.e not
-    # run with 'bazel test ...', only when invoked explicitly
-    # and the "size" set to "enormous".
-    #
-    #"aes_nangate45",
-]]
+]
+
+# Bazel OpenROAD is not compiled with Python support yet
+# leave as manual for now.
+PYTHON_TESTS = [
+    "timing_api",
+    "timing_api_2",
+    "timing_api_3",
+    "timing_api_4",
+    "two_designs",
+]
+
+# TODO There are tests in this folder that were not enabled in
+# CMakeLists.txt, presumably because they are long running.
+# These should be added with tags = ["manual"], i.e not
+# run with 'bazel test ...', only when invoked explicitly
+# and the "size" set to "enormous".
```

Comment:
This is incorrect.  When @QuantamHD converted to using cmake/ctest he didn't convert the flow tests.  They are run by [Jenkins ](https://github.com/The-OpenROAD-Project/OpenROAD/blob/1bd109420d604fdcc8834c7f6379d2128836dd34/Jenkinsfile#L29) and are executed for every PR.

Reply:
`make test` will not run them though which is unfortunate

Reply:
I've tried to correct the comments.

What else is there to do on this PR?

@QuantamHD @hzeller What are your thoughts on local developer workflow expectations vs. CI?

My thinking is that the developer would expect the following to run reasonable set of local tests:

    bazel test -c opt ...

In CI, I don't think it is a big burden to require the test script to craft bazel test statements that picks out tests of a desired range in time and size that are a fit for the available testing resources and testing scope.

Reply:
I think the PR is ready to get submitted, we always can iterate later.

My development expectation would be: `bazel test -c opt ...` should run most test unless they take more than a few minutes in opt mode in which case they should be marked manual (hopefully at some point most *.cc files have a corresponding *_test.cc file with local unit tests to have fast local tests that don't require expensive end2end situations most of the time)

CI should run all the tests (maybe we could even have a configuration for running all the tests, that inclues manual ones as well). And yes, if the CI needs to have a slightly longer command line to accommodate these, that is ok.

Reply:
Today all tests are not run on PR, there for instance some confidential tests. These are are not run on all PRs.

In Bazel it is easy to arrange for things such that stable bazel oneliners implement a policy of which tests to run where in CI when and that these queries will discover the right tests to run without having to update the CI scripts.

Reply:
The fact that they don't run in ctest is not a feature.  Prior to ctest they were run locally as well by the regression script.  It got broken during the move to ctest and nobody has fixed it yet.  I'd like to avoid the same mistake in bazel.  I can merge this as is but don't want to fall into the same later=never trap.

Reply:
Thanks!

I'm confused as to which tests are not available via Bazel.

I have tried to replicate what I conceived as the policy in CMakeLists.txt and is also congruent with expectations from Bazel users(see comments above): fast local tests for developers + longer tests can be run locally if invoked explicitly and from CI.